### ISSUE TITLE
Improve App Interface | App Layer

### ISSUE THREAD

**Issue by GrapesGoober** [GrapesGoober/RSA-App#18]
# Problem
I've taken a peek at your code for the app's interface, and after some thoughts on the chatroom functionality, I've come to some improvements ideas:

## Improved Menu Visuals
This is simple UI change. I just find it hard to read the UI interface properly. Consider using tabs, newlines, and spaces to make it a bit more readable?

## Changing Requirements
After doing some chatroom, I've got a clearer image of what the app should look like. Here's my ideas

### Update Keys
The user can choose to re-generate keys, if they so wish. Input is size (256 bits minimum). The keys will be stored on file. My idea is that the chatroom protocol will auto-sync the keys for us.

### Create New Chatroom
The user should be able to generate new chatrooms. Input: IP and PORT. Preferably, this data is not stored on config file. Once the requirements are inserted, it'll start the chatroom UI

### Connect Chatroom
Before we can connect, there must be an existing chatroom to begin with. Input: IP and PORT. After successful connection, it'll start the chatroom UI.

## Tips
I suggest you try designing a dedicated UI system. Doing hard-coded switch cases like this is very time consuming and very hard to modify later on. Imagine trying to make changes I've asked for above, you'd probably have to rewrite the entire file. Writing more modular code would be better at anticipating future changes and requirements. 

### Functions!!
I see you try to put code into specific functions. You can do better!
```
while True:
   print(""" 
    Welcome to secure TCP chatroom.
      1 - Change Keys
      2 - Create New Chatroom
      3 - Connect Chatroom
      4 - Exit
    Press Number You Wanna Do:
   """)

   match input():
     case "1": change_keys()
     case "2": create_chatroom()
     case "3": connect_chatroom()
     case "4": exit(0)
     case _:   print("invalid inputs")
```
This is just example that you can write just for main menu. Then, for each nested deeper menu, you write dedicated functions inside them. The main menu function exists inside a `while True` loop. Whenever the nested function returns, it'll continue to next loop right away.

Each consecutive loops would have their own UI and input requirements, implemented specifically for their own functions

**Comment by GrapesGoober** [comment]
# Storing Keys In File
In the `config.json` file, there are variables `"e"`, `"d"`, and `"n"`. You can keep those numbers there. Or, more conveniently, store them as `"k_pub"` and `"k_priv"` if you want to. Regardless of your choice, you must be able to read & write the keys from the code. 

Note that the user isn't going to input the keys themselves: they'd just ask the program to regenerate keys and the app will store them there.

**Comment by GrapesGoober** [comment]
# Chatroom unclear
The chatroom class should work now, but I'm considering future changes. The code is ugly, buggy, and generally unpleasant to work with. It should be fine for now, just as a sample API for you, but it might change in the future.

Tomorrow I (should) finish it :)

**Comment by Noravorn2546** [comment]
I add the code to put the keys in the json file but still not have code to get data

I might clear it again tomorrow

**Comment by GrapesGoober** [comment]
# Imma close this for now
Since we're going to work on another issue for file transfer, imma close this as not planned
